osu.Game/Screens/OnlinePlay/Lounge/DrawableLoungeRoom.cs

```diff
@@ -178,46 +179,67 @@ public class PasswordEntryPopover : OsuPopover
         {
             private readonly Room room;
 
-            public Action<Room, string> JoinRequested;
+            public Action<Room, string, Action<Room>, Action<string>> JoinRequested;
 
             public PasswordEntryPopover(Room room)
             {
                 this.room = room;
             }
 
             private OsuPasswordTextBox passwordTextbox;
+            private TriangleButton joinButton;
+            private ShakeContainer shakeContainer;
 
             [BackgroundDependencyLoader]
             private void load()
             {
-                Child = new FillFlowContainer
+                shakeContainer = new ShakeContainer
                 {
                     Margin = new MarginPadding(10),
-                    Spacing = new Vector2(5),
                     AutoSizeAxes = Axes.Both,
-                    Direction = FillDirection.Horizontal,
-                    Children = new Drawable[]
+                    Child = new FillFlowContainer
                     {
-                        passwordTextbox = new OsuPasswordTextBox
+                        Margin = new MarginPadding(10),
+                        Spacing = new Vector2(5),
+                        AutoSizeAxes = Axes.Both,
+                        Direction = FillDirection.Horizontal,
+                        Children = new Drawable[]
                         {
-                            Width = 200,
-                        },
-                        new TriangleButton
-                        {
-                            Width = 80,
-                            Text = "Join Room",
-                            Action = () => JoinRequested?.Invoke(room, passwordTextbox.Text)
+                            passwordTextbox = new OsuPasswordTextBox
+                            {
+                                Width = 200,
+                                PlaceholderText = "password",
+                            },
+                            joinButton = new TriangleButton
+                            {
+                                Width = 80,
+                                Text = "Join Room",
+                            }
                         }
                     }
                 };
+                Child = shakeContainer;
+
+                joinButton.Action = () => JoinRequested?.Invoke(room, passwordTextbox.Text, null, joinFailed);
+            }
+
+            private void joinFailed(string error)
+            {
+                passwordTextbox.Text = string.Empty;
+                passwordTextbox.PlaceholderText = "incorrect password";
```

Comment:
It's pretty weird to do this in the placeholder text I think? Not sure this is even required, the standard login box just shakes.

Reply:
Just shaking the login box instead of giving a response in the placeholder text seems like it fits more with the rest of the lazer UX, so I'll take out the placeholder response.

Reply:
I've removed the placeholder text response but kept the text clear to make it easier for the user to retry.